eeps/eep-0068.md

```diff
@@ -0,0 +1,372 @@
+    Author: Michał Muskała <micmus(at)whatsapp(dot)com>
+    Status: Draft
+    Type: Standards Track
+    Created: 12-02-2024
+    Erlang-Version:
+    Post-History:
+****
+# EEP 68: JSON library
+----
+
+## Abstract
+
+This EEP proposes introducing a module `json` to the Erlang standard
+library with support for encoding and decoding [JSON][1] documents
+from and to Erlang data structures. The main reason is to cover
+a gap in the Erlang standard library with regards to such a vastly
+popular and widespread data format.
+
+## Rationale
+
+JSON is commonly used in many different use-cases:
+* by web services as a lightweight and human-readable data interchange format;
+* as a configuration language in static files;
+* as data interchange format by developer tooling;
+* and more.
+
+There are many existing JSON libraries for Erlang and other BEAM languages,
+however adding such a support to standard library would offer unique benefits.
+Most notably being able to use it in situations where leveraging third-party
+libraries is complex or cumbersome -- such as stand-alone escripts or
+fundamental tooling like a build system, or inside OTP itself.
+
+There have been previous attempts to bring JSON support into OTP, most notably
+[EEP 18][EEP], which ultimately weren't adopted previously for various reasons.
+However, I believe the time is right to revisit this subject with a fresh
+take on an interface such support could take.
```

Comment:
From Ericsson's point of view, I believe the key difference as to why JSON can be acceptable in OTP now and not in the past, is the introduction of JSON in the 5G network standards.

As opposed to earlier mobile networks (2G, 3G, 4G), the 5G network uses JSON REST-APIs over HTTP/2 in what is called "Service Based Architecture". The specifications are open but it's quite difficult to find the right ones. 3GPP TS 29.500 is a good starting point for this though: https://www.etsi.org/deliver/etsi_ts/129500_129599/129500/16.04.00_60/ts_129500v160400p.pdf

The the older 2G, 3G and 4G networks there was no use of JSON or TCP or HTTP or almost any web protocols. There were telecom-specific protocols (such as DIAMETER) for everything.